pkg/kv/txn_interceptor_pipeliner.go

```diff
@@ -425,32 +429,32 @@ func (tp *txnPipeliner) chainToInFlightWrites(ba roachpb.BatchRequest) roachpb.B
 	return ba
 }
 
-// updateWriteTracking reads the response for the given request and uses it to
-// update the tracked in-flight write set and write footprint. It does so by
-// performing three actions: 1. it adds all async writes that the request
-// performed to the in-flight
-//    write set.
-// 2. it adds all non-async writes that the request performed to the write
-//    footprint.
-// 3. it moves all in-flight writes that the request proved to exist from
-//    the in-flight writes set to the write footprint.
+// updateLockTracking reads the response for the given request and uses it to
+// update the tracked in-flight write set and lock footprint. It does so by
+// performing three actions:
+//  1. it adds all async writes that the request performed to the in-flight
+//     write set.
+//  2. it adds all non-async writes and locking reads that the request
+//     performed to the lock footprint.
+//  3. it moves all in-flight writes that the request proved to exist from
+//     the in-flight writes set to the lock footprint.
 //
-// After updating the write sets, the write footprint is condensed to ensure
-// that it remains under its memory limit.
+// After updating the write sets, the lock footprint is condensed to ensure that
+// it remains under its memory limit.
```

Comment:
Ah, now I understand. The footprint only has spans, it doesn't care whether a lock is replicated or not. It's just for cleanup.